File: MATLAB/ProjectEuler_07.m
%% 10001st Prime -- Solved
clear, close, clc, disp(mfilename), tic;

%% Solution 1: Use primes() Function
n = primes(10001^2);
Solution = n(10001);

%% Solution 2:
% n = [1];
% Counter = 1;
% 
% for i = 1:10001^2
%     if isprime(i) == true
%         n(Counter) = i;
%         Counter = Counter + 1;
%     end
% 
%     if length(n) == 10001
%         break;
%     end
% end
% 
% Solution = n(end);

%% Solution 3: -- Incomplete
% 
% n = 1:10001^2;
% p = [];
% Counter = 1;
% 
% for i = 2:10001^2/2
%     for j = 2:10001^2/2
%         if n(i) ~= n(j) && rem(n(i),j) == 0
%             p(Counter) = n(i);
%             Counter = Counter + 1;
%         end
%         
%         if length(p) == 10001
%             break;
%         end
%     end
% end
%     
% Solution = p(end);

%% Display Solution
fprintf('The 10001st prime number is: %d\n', Solution); toc;

File: Python/ProjectEuler_07.py
# 1001st Prime -- Solved
# Solution = 104743

import math

from Timer import Timer
Stopwatch = Timer()

def PrimeCheck(n):

    if n <= 1:
        return None

    if n % 2 == 0 and n != 2:
        return False
    elif n == 2:
        return True

    Factor = 3
    while Factor < round(math.sqrt(n)) + 1:
        if n % Factor == 0:
            if n != Factor:
                return False
        Factor += 2

    return True

Primes = [2]
n = 3

while len(Primes) <= 10001:
    if PrimeCheck(n):
        Primes.append(n)

    n += 2

Solution = Primes[10000]

print('The 10001st prime number is: ', Solution)
Stopwatch.stop()
